cli/smartcontract/contract_test.go

```diff
@@ -213,6 +213,110 @@ func ToMap(a []testcontract.MyPair) map[int]string {
 `, string(bs))
 }
 
+func TestDynamicWrapper(t *testing.T) {
+	smartcontract.ModVersion = "v0.0.0"
+
+	tmpDir := t.TempDir()
+	e := testcli.NewExecutor(t, true)
+
+	ctrPath := filepath.Join(tmpDir, "testcontract")
+	e.Run(t, "neo-go", "contract", "init", "--name", ctrPath)
+
+	cfgPath := filepath.Join(ctrPath, "neo-go.yml")
+	bindingsPath := filepath.Join(tmpDir, "bindings.yml")
+	manifestPath := filepath.Join(tmpDir, "manifest.json")
+	cmd := []string{"neo-go", "contract", "compile", "--in", ctrPath}
+
+	goMod := filepath.Join(ctrPath, "go.mod")
+	data, err := os.ReadFile(goMod)
+	require.NoError(t, err)
+
+	i := bytes.IndexByte(data, '\n')
+	data = append([]byte("module myimport.com/testcontract"), data[i:]...)
+
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+	data = append(data, "\nreplace github.com/nspcc-dev/neo-go/pkg/interop => "...)
+	data = append(data, filepath.Join(wd, "../../pkg/interop")...)
+	require.NoError(t, os.WriteFile(goMod, data, os.ModePerm))
+
+	cmd = append(cmd, "--config", cfgPath,
+		"--out", filepath.Join(tmpDir, "out.nef"),
+		"--manifest", manifestPath,
+		"--bindings", bindingsPath)
+
+	e.Run(t, cmd...)
+	e.CheckEOF(t)
+	require.FileExists(t, bindingsPath)
+
+	deployedConctract := testcli.DeployContract(t, e, filepath.Join(ctrPath, "main.go"), filepath.Join(ctrPath, "neo-go.yml"), testcli.ValidatorWallet, testcli.ValidatorAddr, testcli.ValidatorPass)
+
+	helperContract := `package helpercontract
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop"
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+)
+
+func Main(operation string, args []interface{}) interface{} {
+	switch operation {
+	case "callMainContract":
+		if len(args) != 1 {
+			runtime.Log("Incorrect number of arguments")
+			return false
+		}
+		hash, ok := args[0].(interop.Hash160)
+		if !ok {
+			runtime.Log("Argument is not a valid Hash160")
+			return false
+		}
+		callMainContract(hash)
+		return true
+	default:
+		runtime.Log("Unsupported operation")
+		return false
+	}
+}
+
+func callMainContract(hash interop.Hash160) {
+	contractInstance := NewContract(hash)
+
+	contractInstance.RuntimeNotify([]any{"Example argument"})
+}`
```

Comment:
Don't need to place it to a separate function. Move this code to `CallVerifyContract`.

It won't work with missing import. You need something like:
```
contractInstance := package.NewContract(hash)
return contractInstance.Verify()
```

where `package` is your binding package. You also have to create proper go.mod for the helper contract to be able to compile it.

Reply:
The compile/deploy part may be done via `testcli.DeployContract` helper.